src/main/java/com/datadoghq/datadog_lambda_java/DDLambda.java

```diff
@@ -1,150 +1,217 @@
+        addTraceContextToMDC();
+        addTagsToSpan(cxt);
     }
 
     /**
      * Create a trace-enabled DDLambda instrumenter given an APIGatewayProxyRequestEvent and a Lambda context
+     *
      * @param req Your Datadog trace headers are pulled from the request and sent to XRay for consumption by the
      *            Datadog Xray Crawler
      * @param cxt Enhanced Metrics pulls information from the Lambda context.
      */
-    public DDLambda(APIGatewayProxyRequestEvent req, Context cxt){
+    public DDLambda(APIGatewayProxyRequestEvent req, Context cxt) {
         this.enhanced = checkEnhanced();
         recordEnhanced(INVOCATION, cxt);
         this.tracing = new Tracing(req);
         this.tracing.submitSegment();
-        MDC.put(MDC_TRACE_CONTEXT_FIELD, getTraceContextString());
+        addTraceContextToMDC();
+        addTagsToSpan(cxt);
     }
 
     /**
      * Create a trace-enabled DDLambda instrumenter given an APIGatewayV2ProxyEventRequest and a Lambda context
+     *
      * @param req Your Datadog trace headers are pulled from the request and sent to XRay for consumption by the
      *            Datadog Xray Crawler
      * @param cxt Enhanced Metrics pulls information from the Lambda context.
      */
-    public DDLambda(APIGatewayV2ProxyRequestEvent req, Context cxt){
+    public DDLambda(APIGatewayV2ProxyRequestEvent req, Context cxt) {
         this.enhanced = checkEnhanced();
         recordEnhanced(INVOCATION, cxt);
         this.tracing = new Tracing(req);
         this.tracing.submitSegment();
-        MDC.put(MDC_TRACE_CONTEXT_FIELD, getTraceContextString());
+        addTraceContextToMDC();
+        addTagsToSpan(cxt);
     }
 
     /**
      * Create a trace-enabled DDLambda instrumenter given a custom request object. Please note that your custom request
      * object MUST implement Headerable.
+     *
      * @param req A custom request object that implements Headerable. Datadog trace headers are pulled from this request object.
      * @param cxt Enhanced Metrics pulls information from the Lambda context.
      */
-    public DDLambda(Headerable req, Context cxt){
+    public DDLambda(Headerable req, Context cxt) {
         this.enhanced = checkEnhanced();
         recordEnhanced(INVOCATION, cxt);
         this.tracing = new Tracing(req);
         this.tracing.submitSegment();
+        addTraceContextToMDC();
+        addTagsToSpan(cxt);
+    }
+
+    private void addTagsToSpan(Context cxt) {
+        String requestId = "";
+        String functionName = "";
+        String functionArn = "";
+        if (cxt != null) {
+            requestId = cxt.getAwsRequestId();
+            functionName = cxt.getFunctionName();
+            functionArn = santitizeFunctionArn(cxt.getInvokedFunctionArn());
+        }
+        Span span = GlobalTracer.get().activeSpan();
+        if (span != null) {
+            span.setTag("request_id", requestId);
```

Comment:
I think the other libraries add `function_version` and `resource_names` tags as well.

Reply:
👍